### ISSUE TITLE
The shape of logit in loss function

### ISSUE THREAD

**Issue by hungthanhpham94** [yhs-968/pyGRU4REC#5]
Dear Younghun Song,
First of all, I really thank you for your useful implementation. It is simpler than original code except a point. In function BPRLoss, I don't understand why the shape of logit is BxB. I think that it should be B x n_item.
Can you explain it to me?
Thank you for your help :)

**Comment by yhs-968** [comment]
Sorry for the late reply. I understand that the loss function is quite confusing - it is a computational trick for a speed-up.

The `logit` parameter of the `BPRloss` refers to the **sampled logit** that only contains the scores for the **negative items** in a session-parallel minibatch, as mentioned in the original GRU4REC paper. It's related to the line 112-115 of [this module](https://github.com/yhs-968/pyGRU4REC/blob/master/modules/model.py).

Given the sampled logit, `BPRLoss` behaves as if the items corresponding to the sampled logit is equal to the entire item basket, which is totally fine since the loss is only calculated with respect to the items in a session-parallel minibatch.

**Comment by hungthanhpham94** [comment]
Oh Yes, You don't need to apologize :D. When I read your replies in other issues, I knew that you are busy.  :D 
Your answer is very clear to me. Session-parallel mini batching was not familiar to me, and therefore I did not know what negative items are :D. So, I thought that **N_s** (the sample size) in paper is the number of items :D 
Thank you very much :D
